projects/angular-auth-oidc-client/src/lib/interceptor/auth.interceptor.spec.ts

```diff
@@ -15,42 +15,78 @@ describe(`AuthHttpInterceptor`, () => {
-        {
-          provide: LoggerService,
-          useClass: mockClass(LoggerService),
-        },
-        {
-          provide: ConfigurationService,
-          useClass: mockClass(ConfigurationService),
-        },
-      ],
+  describe(`with Class Interceptor`, () => {
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [HttpClientTestingModule],
+        providers: [
+          ClosestMatchingRouteService,
+          {
+            provide: HTTP_INTERCEPTORS,
+            useClass: AuthInterceptor,
+            multi: true,
+          },
+          { provide: AuthStateService, useClass: mockClass(AuthStateService) },
+          {
+            provide: LoggerService,
+            useClass: mockClass(LoggerService),
+          },
+          {
+            provide: ConfigurationService,
+            useClass: mockClass(ConfigurationService),
+          },
+        ],
+      });
+
+      httpClient = TestBed.inject(HttpClient);
+      httpTestingController = TestBed.inject(HttpTestingController);
+      configurationService = TestBed.inject(ConfigurationService);
+      authStateService = TestBed.inject(AuthStateService);
+      closestMatchingRouteService = TestBed.inject(ClosestMatchingRouteService);
+    });
+
+    afterEach(() => {
+      httpTestingController.verify();
     });
 
-    httpClient = TestBed.inject(HttpClient);
-    httpTestingController = TestBed.inject(HttpTestingController);
-    configurationService = TestBed.inject(ConfigurationService);
-    authStateService = TestBed.inject(AuthStateService);
-    closestMatchingRouteService = TestBed.inject(ClosestMatchingRouteService);
+    runTests();
   });
 
-  afterEach(() => {
-    httpTestingController.verify();
+  describe(`with Functional Interceptor`, () => {
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        providers: [
+          ClosestMatchingRouteService,
+          provideHttpClient(withInterceptors([authInterceptor()])),
+          provideHttpClientTesting(),
+          { provide: AuthStateService, useClass: mockClass(AuthStateService) },
+          {
+            provide: LoggerService,
+            useClass: mockClass(LoggerService),
+          },
+          {
+            provide: ConfigurationService,
+            useClass: mockClass(ConfigurationService),
+          },
+        ],
+      });
+
+      httpClient = TestBed.inject(HttpClient);
+      httpTestingController = TestBed.inject(HttpTestingController);
+      configurationService = TestBed.inject(ConfigurationService);
+      authStateService = TestBed.inject(AuthStateService);
+      closestMatchingRouteService = TestBed.inject(ClosestMatchingRouteService);
+    });
+
+    afterEach(() => {
+      httpTestingController.verify();
+    });
+
+    runTests();
   });
 
-  it(
-    'should add an Authorization header when route matches and token is present',
-    waitForAsync(() => {
+  function runTests() {
```

Comment:
The same tests are run for the class and functional interceptors.
The only difference is the setup in the `beforeEach` block.